Makefile

```diff
@@ -30,37 +30,24 @@ venv:  ## Build Python virtual environment
 requirements: metadeps installreqs  ## Install/refresh Python project requirements including those needed for development
 
 .PHONY: updatereqs
-updatereqs:  ## Auto update and generate requirements.txt
-	@echo "Upgrading pip..."
-	$(VENV_BIN)/python -m pip install --upgrade pip
-	$(VENV_BIN)/pip install -U pip-tools
-	-@rm python/quantum-pecos/requirements.txt
-	@echo "Using version: $(call get_version)"
-	@echo "Temporarily modifying pyproject.toml..."
-	@sed -i.bak '/pecos-rslib==/d' $(PYPROJECT_PATH)
-	$(VENV_BIN)/pip-compile --extra=tests --no-annotate --no-emit-index-url \
-		--output-file=python/quantum-pecos/requirements.txt \
-		--strip-extras $(PYPROJECT_PATH)
-	@echo "Restoring original pyproject.toml..."
-	@mv $(PYPROJECT_PATH).bak $(PYPROJECT_PATH)
-	@echo "Adding pecos-rslib back to requirements.txt..."
-	@echo "pecos-rslib==$(call get_version)" >> python/quantum-pecos/requirements.txt
-	@echo "numpy==1.26.4 ; python_version < \"3.13\"" >> python/quantum-pecos/requirements.txt
+updatereqs:  ## Generate/update lockfiles for both packages
+	@echo "Upgrading uv..."
+	$(VENV_BIN)/python -m pip install --upgrade uv
+	@echo "Generating lockfiles..."
+	cd python/pecos-rslib && $(VENV_BIN)/uv lock
+	cd python/quantum-pecos && $(VENV_BIN)/uv lock
 
 .PHONY: installreqs
-installreqs: upgrade-pip ## Install Python project requirements
-	@echo "Temporarily removing pecos-rslib from requirements..."
-	@grep -v "pecos-rslib" python/quantum-pecos/requirements.txt > temp_requirements.txt
-
-	@echo "Installing project requirements (excluding pecos-rslib)..."
-	$(VENV_BIN)/pip install -r temp_requirements.txt
-
-	@echo "Cleaning up temporary files..."
-	@rm temp_requirements.txt
+installreqs: ## Install Python project requirements from lockfiles
+	@unset CONDA_PREFIX \
+	&& $(VENV_BIN)/python -m pip install --upgrade uv
+	@echo "Installing from lockfiles..."
+	cd python/pecos-rslib && $(VENV_BIN)/uv sync
+	cd python/quantum-pecos && $(VENV_BIN)/uv sync
```

Comment:
If you are using `uv` then it should not be installed inside a virtual environment, instead it should be installed globally on a system (ie, README should say it's required). It handles the virtual environment for you and can even handle python versions.

Then to get the project ready, all one needs to do is `uv sync` assuming all dependencies are correctly specified in `pyproject.toml`.

Reply:
Ah just saw this. Yeah, I think I happened to arrive at the same conclusion. Switching this to a draft PR

Reply:
hmmm, looking through things I see more `$(VENV_BIN)/` that I can get rid of using `uvx` although I seem to be having issues getting `pytest` to work. :thinking:

Reply:
NVM, `uv run pytest` works and `uvx pytest` doesn't for some reason.

Reply:
`uvx` shouldn't generally be needed for a project. Its primary goal is to be able to run a python command that's not even installed on the system, along the lines of [pipx](https://github.com/pypa/pipx). Only useful for system utilities or rare invocations of such programs, IMHO.

Reply:
Makes sense. Reading up on `uvx`, it sounds like it creates an isolated environment to run a command while `uv run` uses your venv. So makes sense that what you say is best practice. thanks!